hibernate4/src/main/java/com/fasterxml/jackson/datatype/hibernate4/HibernateProxySerializer.java

```diff
@@ -194,4 +200,40 @@ protected Object findProxied(HibernateProxy proxy)
         }
         return init.getImplementation();
     }
+    
+    // Alas, hibernate offers no public api to access this information, so we must resort to ugly hacks ...
+    protected static class ProxyReader {
+
+        // static final so the JVM can inline the lookup
+        private static final Field getIdentifierMethod;
+
+        static {
+            try {
+                getIdentifierMethod = BasicLazyInitializer.class.getDeclaredField("getIdentifierMethod");
+                getIdentifierMethod.setAccessible(true);
+            } catch (Exception e) {
+            	// should never happen: the field exists in all versions of hibernate 4 and 5
+                throw new RuntimeException(e); 
+            }
+        }
+
+        /**
+         * @return the name of the identifier property, or null if the name could not be determined
+         */
+        static String getIdentifierPropertyName(LazyInitializer init) {
+            try {
+                Method idGetter = (Method) getIdentifierMethod.get(init);
+                if (idGetter == null) {
+                	return null;
+                }
+                String name = idGetter.getName();
+                if (name.startsWith("get")) {
```

Comment:
Probably has to remain here; ideally `BeanUtil` from `jackson-databind` could be used but unfortunately `okNameForIsGetter` requires `AnnotatedMethod`, not base `Method` so it can not be used.

But there is `Introspector.decapitalize()`:

http://docs.oracle.com/javase/7/docs/api/java/beans/Introspector.html#decapitalize(java.lang.String)

which is what you can call, so:

    String name = Introspector.decapitalize(name.substring(3));

if name starts with "get". And the reason to call the method is that `getURL()` should result in `URL`, and NOT `uRL`.